### DISCUSSION TITLE
Cannot read properties of undefined (reading ‘data’)

### DISCUSSION THREAD

**Discussion by YoKONCy** [lianginx/chatgpt-nuxt#4] [General]
我在本地跑了之后，然后填入了我自己的秘钥，发送了一条信息，然后显示Cannot read properties of undefined (reading ‘data’)，这是啥

**Comment by lianginx** [comment]
500 是指服务器端有错误，经过我的测试，这应该是你的网络环境有问题，导致服务端代码中请求 OpenAI API 的代码连接超时了。

API 请求需要走梯子的代理，或者直接将项目运行在可以直连 OpenAI 的服务器上就不会出现这个问题了。

**Reply by lianginx** [reply]
如何你用的 Clash，可以打开 TUN Mode 来进行全局代理，具体教程参考[官方文档](https://docs.cfw.lbyczf.com/contents/tun.html)

**Comment by YoKONCy** [comment]
问题是我服务器在美丽国，应该不是梯子的问题，然后执行npm install 的时候，报了错，好像是有两个插件需要的vue版本不一样，然后我就强制安装了

**Reply by lianginx** [reply]
依赖版本的问题，我一直用的 yarn，它好像自动处理了这些错误，所以我一直没有注意到。

`data undefined` 报错是因为我在 `/server/api/chat.ts` 里捕获异常的代码写的有问题，所以报错了，而且没有把实际报错信息抛到前端，今天晚上我修复一下。

然后我找了一台Windows电脑克隆项目运行了一下，好像复现了你的问题，报错 `data undefined`，然后我打印了真实的报错信息，提示：`Client network socket disconnected before secure TLS connection was established`

在我还没搞明白这是什么意思的时候，我再次发送请求测试，发现它又正常了，因为我测试中一般后端代码报错都是请求 OpenAI API 时网络异常，所以我怀疑还是网络有问题……

**Comment by lianginx** [comment]
有人遇到了同样的问题，我分别回复不太方便，如果你还有问题请到这里 https://github.com/lianginx/chatgpt-nuxt/issues/5 讨论，我们一起解决这个问题，感谢！
